slasher/db/kv/spanner_new.go

```diff
@@ -0,0 +1,102 @@
+package kv
+
+import (
+	"context"
+	"errors"
+
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/slasher/detection/attestations/types"
+	bolt "go.etcd.io/bbolt"
+	"go.opencensus.io/trace"
+)
+
+var errWrongSize = errors.New("wrong data length for min max span byte array")
+var highestObservedValidatorIdx uint64
+
+// GetValidatorSpan unmarshal a span from an encoded, flattened array.
+func (es EpochStore) GetValidatorSpan(ctx context.Context, idx uint64) (types.Span, error) {
+	r := types.Span{}
+	if len(es)%spannerEncodedLength != 0 {
+		return r, errWrongSize
+	}
+	origLength := uint64(len(es)) / spannerEncodedLength
+	requestedLength := idx + 1
+	if origLength < requestedLength {
+		return r, nil
+	}
+	cursor := idx * spannerEncodedLength
+	r.MinSpan = bytesutil.FromBytes2(es[cursor : cursor+2])
+	r.MaxSpan = bytesutil.FromBytes2(es[cursor+2 : cursor+4])
+	sigB := [2]byte{}
+	copy(sigB[:], es[cursor+4:cursor+6])
+	r.SigBytes = sigB
+	r.HasAttested = bytesutil.ToBool(es[cursor+6])
+	return r, nil
+}
+
+// SetValidatorSpan marshal a validator span into an encoded, flattened array.
+func (es EpochStore) SetValidatorSpan(ctx context.Context, idx uint64, newSpan types.Span) error {
+	if len(es)%spannerEncodedLength != 0 {
+		return errors.New("wrong data length for min max span byte array")
```

Comment:
Maybe this can be moved into a `NewEpochStore([]byte)` function? Seems weird to have the function return an error for something that should be checked on creation.

Would be great to make an `epoch_store.go` with these functions as well.

Reply:
added NewEpochStore. kept error in place for extra safety